// @kinde-oss/kinde-nodejs-sdk/dist/index.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
Object.defineProperty(exports, "AddOrganizationUsersRequest", {
  enumerable: true,
  get: function get() {
    return _AddOrganizationUsersRequest["default"];
  }
});
Object.defineProperty(exports, "AddOrganizationUsersRequestUsersInner", {
  enumerable: true,
  get: function get() {
    return _AddOrganizationUsersRequestUsersInner["default"];
  }
});
Object.defineProperty(exports, "AddOrganizationUsersResponse", {
  enumerable: true,
  get: function get() {
    return _AddOrganizationUsersResponse["default"];
  }
});
Object.defineProperty(exports, "ApiClient", {
  enumerable: true,
  get: function get() {
    return _ApiClient["default"];
  }
});
Object.defineProperty(exports, "ApiResult", {
  enumerable: true,
  get: function get() {
    return _ApiResult["default"];
  }
});
Object.defineProperty(exports, "Application", {
  enumerable: true,
  get: function get() {
    return _Application["default"];
  }
});
Object.defineProperty(exports, "AuthorizationCode", {
  enumerable: true,
  get: function get() {
    return _AuthorizationCode["default"];
  }
});
Object.defineProperty(exports, "CallbacksApi", {
  enumerable: true,
  get: function get() {
    return _CallbacksApi["default"];
  }
});
Object.defineProperty(exports, "ClientCredentials", {
  enumerable: true,
  get: function get() {
    return _ClientCredentials["default"];
  }
});
Object.defineProperty(exports, "ConnectedAppsAccessToken", {
  enumerable: true,
  get: function get() {
    return _ConnectedAppsAccessToken["default"];
  }
});
Object.defineProperty(exports, "ConnectedAppsApi", {
  enumerable: true,
  get: function get() {
    return _ConnectedAppsApi["default"];
  }
});
Object.defineProperty(exports, "ConnectedAppsAuthUrl", {
  enumerable: true,
  get: function get() {
    return _ConnectedAppsAuthUrl["default"];
  }
});
Object.defineProperty(exports, "CreateOrganizationRequest", {
  enumerable: true,
  get: function get() {
    return _CreateOrganizationRequest["default"];
  }
});
Object.defineProperty(exports, "CreateOrganizationResponse", {
  enumerable: true,
  get: function get() {
    return _CreateOrganizationResponse["default"];
  }
});
Object.defineProperty(exports, "CreateOrganizationResponseOrganization", {
  enumerable: true,
  get: function get() {
    return _CreateOrganizationResponseOrganization["default"];
  }
});
Object.defineProperty(exports, "CreateOrganizationUserRoleRequest", {
  enumerable: true,
  get: function get() {
    return _CreateOrganizationUserRoleRequest["default"];
  }
});
Object.defineProperty(exports, "CreatePermissionRequest", {
  enumerable: true,
  get: function get() {
    return _CreatePermissionRequest["default"];
  }
});
Object.defineProperty(exports, "CreateRoleRequest", {
  enumerable: true,
  get: function get() {
    return _CreateRoleRequest["default"];
  }
});
Object.defineProperty(exports, "CreateUserRequest", {
  enumerable: true,
  get: function get() {
    return _CreateUserRequest["default"];
  }
});
Object.defineProperty(exports, "CreateUserRequestIdentitiesInner", {
  enumerable: true,
  get: function get() {
    return _CreateUserRequestIdentitiesInner["default"];
  }
});
Object.defineProperty(exports, "CreateUserRequestIdentitiesInnerDetails", {
  enumerable: true,
  get: function get() {
    return _CreateUserRequestIdentitiesInnerDetails["default"];
  }
});
Object.defineProperty(exports, "CreateUserRequestProfile", {
  enumerable: true,
  get: function get() {
    return _CreateUserRequestProfile["default"];
  }
});
Object.defineProperty(exports, "CreateUserResponse", {
  enumerable: true,
  get: function get() {
    return _CreateUserResponse["default"];
  }
});
Object.defineProperty(exports, "EnvironmentsApi", {
  enumerable: true,
  get: function get() {
    return _EnvironmentsApi["default"];
  }
});
Object.defineProperty(exports, "Error", {
  enumerable: true,
  get: function get() {
    return _Error["default"];
  }
});
Object.defineProperty(exports, "ErrorResponse", {
  enumerable: true,
  get: function get() {
    return _ErrorResponse["default"];
  }
});
Object.defineProperty(exports, "FeatureFlagsApi", {
  enumerable: true,
  get: function get() {
    return _FeatureFlagsApi["default"];
  }
});
Object.defineProperty(exports, "GetApplicationsResponse", {
  enumerable: true,
  get: function get() {
    return _GetApplicationsResponse["default"];
  }
});
Object.defineProperty(exports, "GetOrganizationsResponse", {
  enumerable: true,
  get: function get() {
    return _GetOrganizationsResponse["default"];
  }
});
Object.defineProperty(exports, "GetOrganizationsUserRolesResponse", {
  enumerable: true,
  get: function get() {
    return _GetOrganizationsUserRolesResponse["default"];
  }
});
Object.defineProperty(exports, "GetOrganizationsUsersResponse", {
  enumerable: true,
  get: function get() {
    return _GetOrganizationsUsersResponse["default"];
  }
});
Object.defineProperty(exports, "GetRedirectCallbackUrlsResponse", {
  enumerable: true,
  get: function get() {
    return _GetRedirectCallbackUrlsResponse["default"];
  }
});
Object.defineProperty(exports, "GrantType", {
  enumerable: true,
  get: function get() {
    return _GrantType["default"];
  }
});
Object.defineProperty(exports, "KindeClient", {
  enumerable: true,
  get: function get() {
    return _KindeClient["default"];
  }
});
Object.defineProperty(exports, "OAuthApi", {
  enumerable: true,
  get: function get() {
    return _OAuthApi["default"];
  }
});
Object.defineProperty(exports, "Organization", {
  enumerable: true,
  get: function get() {
    return _Organization["default"];
  }
});
Object.defineProperty(exports, "OrganizationUser", {
  enumerable: true,
  get: function get() {
    return _OrganizationUser["default"];
  }
});
Object.defineProperty(exports, "OrganizationUserRole", {
  enumerable: true,
  get: function get() {
    return _OrganizationUserRole["default"];
  }
});
Object.defineProperty(exports, "OrganizationsApi", {
  enumerable: true,
  get: function get() {
    return _OrganizationsApi["default"];
  }
});
Object.defineProperty(exports, "PKCE", {
  enumerable: true,
  get: function get() {
    return _PKCE["default"];
  }
});
Object.defineProperty(exports, "Permissions", {
  enumerable: true,
  get: function get() {
    return _Permissions["default"];
  }
});
Object.defineProperty(exports, "PermissionsApi", {
  enumerable: true,
  get: function get() {
    return _PermissionsApi["default"];
  }
});
Object.defineProperty(exports, "RedirectCallbackUrls", {
  enumerable: true,
  get: function get() {
    return _RedirectCallbackUrls["default"];
  }
});
Object.defineProperty(exports, "RefreshToken", {
  enumerable: true,
  get: function get() {
    return _RefreshToken["default"];
  }
});
Object.defineProperty(exports, "RemoveOrganizationUsersResponse", {
  enumerable: true,
  get: function get() {
    return _RemoveOrganizationUsersResponse["default"];
  }
});
Object.defineProperty(exports, "Roles", {
  enumerable: true,
  get: function get() {
    return _Roles["default"];
  }
});
Object.defineProperty(exports, "RolesApi", {
  enumerable: true,
  get: function get() {
    return _RolesApi["default"];
  }
});
Object.defineProperty(exports, "SessionStore", {
  enumerable: true,
  get: function get() {
    return _SessionStore["default"];
  }
});
Object.defineProperty(exports, "SuccessResponse", {
  enumerable: true,
  get: function get() {
    return _SuccessResponse["default"];
  }
});
Object.defineProperty(exports, "UpdateOrganizationRequest", {
  enumerable: true,
  get: function get() {
    return _UpdateOrganizationRequest["default"];
  }
});
Object.defineProperty(exports, "UpdateOrganizationUsersRequest", {
  enumerable: true,
  get: function get() {
    return _UpdateOrganizationUsersRequest["default"];
  }
});
Object.defineProperty(exports, "UpdateOrganizationUsersRequestUsersInner", {
  enumerable: true,
  get: function get() {
    return _UpdateOrganizationUsersRequestUsersInner["default"];
  }
});
Object.defineProperty(exports, "UpdateUserRequest", {
  enumerable: true,
  get: function get() {
    return _UpdateUserRequest["default"];
  }
});
Object.defineProperty(exports, "User", {
  enumerable: true,
  get: function get() {
    return _User["default"];
  }
});
Object.defineProperty(exports, "UserIdentity", {
  enumerable: true,
  get: function get() {
    return _UserIdentity["default"];
  }
});
Object.defineProperty(exports, "UserIdentityResult", {
  enumerable: true,
  get: function get() {
    return _UserIdentityResult["default"];
  }
});
Object.defineProperty(exports, "UserProfile", {
  enumerable: true,
  get: function get() {
    return _UserProfile["default"];
  }
});
Object.defineProperty(exports, "UserProfileV2", {
  enumerable: true,
  get: function get() {
    return _UserProfileV["default"];
  }
});
Object.defineProperty(exports, "UsersApi", {
  enumerable: true,
  get: function get() {
    return _UsersApi["default"];
  }
});
Object.defineProperty(exports, "UsersResponse", {
  enumerable: true,
  get: function get() {
    return _UsersResponse["default"];
  }
});
var _ApiClient = _interopRequireDefault(require("./ApiClient"));
var _AuthorizationCode = _interopRequireDefault(require("./sdk/oauth2/AuthorizationCode"));
var _ClientCredentials = _interopRequireDefault(require("./sdk/oauth2/ClientCredentials"));
var _GrantType = _interopRequireDefault(require("./sdk/constant/GrantType"));
var _KindeClient = _interopRequireDefault(require("./KindeClient"));
var _PKCE = _interopRequireDefault(require("./sdk/oauth2/PKCE"));
var _RefreshToken = _interopRequireDefault(require("./sdk/oauth2/RefreshToken"));
var _SessionStore = _interopRequireDefault(require("./sdk/store/SessionStore"));
var _AddOrganizationUsersRequest = _interopRequireDefault(require("./model/AddOrganizationUsersRequest"));
var _AddOrganizationUsersRequestUsersInner = _interopRequireDefault(require("./model/AddOrganizationUsersRequestUsersInner"));
var _AddOrganizationUsersResponse = _interopRequireDefault(require("./model/AddOrganizationUsersResponse"));
var _ApiResult = _interopRequireDefault(require("./model/ApiResult"));
var _Application = _interopRequireDefault(require("./model/Application"));
var _ConnectedAppsAccessToken = _interopRequireDefault(require("./model/ConnectedAppsAccessToken"));
var _ConnectedAppsAuthUrl = _interopRequireDefault(require("./model/ConnectedAppsAuthUrl"));
var _CreateOrganizationRequest = _interopRequireDefault(require("./model/CreateOrganizationRequest"));
var _CreateOrganizationResponse = _interopRequireDefault(require("./model/CreateOrganizationResponse"));
var _CreateOrganizationResponseOrganization = _interopRequireDefault(require("./model/CreateOrganizationResponseOrganization"));
var _CreateOrganizationUserRoleRequest = _interopRequireDefault(require("./model/CreateOrganizationUserRoleRequest"));
var _CreatePermissionRequest = _interopRequireDefault(require("./model/CreatePermissionRequest"));
var _CreateRoleRequest = _interopRequireDefault(require("./model/CreateRoleRequest"));
var _CreateUserRequest = _interopRequireDefault(require("./model/CreateUserRequest"));
var _CreateUserRequestIdentitiesInner = _interopRequireDefault(require("./model/CreateUserRequestIdentitiesInner"));
var _CreateUserRequestIdentitiesInnerDetails = _interopRequireDefault(require("./model/CreateUserRequestIdentitiesInnerDetails"));
var _CreateUserRequestProfile = _interopRequireDefault(require("./model/CreateUserRequestProfile"));
var _CreateUserResponse = _interopRequireDefault(require("./model/CreateUserResponse"));
var _Error = _interopRequireDefault(require("./model/Error"));
var _ErrorResponse = _interopRequireDefault(require("./model/ErrorResponse"));
var _GetApplicationsResponse = _interopRequireDefault(require("./model/GetApplicationsResponse"));
var _GetOrganizationsResponse = _interopRequireDefault(require("./model/GetOrganizationsResponse"));
var _GetOrganizationsUserRolesResponse = _interopRequireDefault(require("./model/GetOrganizationsUserRolesResponse"));
var _GetOrganizationsUsersResponse = _interopRequireDefault(require("./model/GetOrganizationsUsersResponse"));
var _GetRedirectCallbackUrlsResponse = _interopRequireDefault(require("./model/GetRedirectCallbackUrlsResponse"));
var _Organization = _interopRequireDefault(require("./model/Organization"));
var _OrganizationUser = _interopRequireDefault(require("./model/OrganizationUser"));
var _OrganizationUserRole = _interopRequireDefault(require("./model/OrganizationUserRole"));
var _Permissions = _interopRequireDefault(require("./model/Permissions"));
var _RedirectCallbackUrls = _interopRequireDefault(require("./model/RedirectCallbackUrls"));
var _RemoveOrganizationUsersResponse = _interopRequireDefault(require("./model/RemoveOrganizationUsersResponse"));
var _Roles = _interopRequireDefault(require("./model/Roles"));
var _SuccessResponse = _interopRequireDefault(require("./model/SuccessResponse"));
var _UpdateOrganizationRequest = _interopRequireDefault(require("./model/UpdateOrganizationRequest"));
var _UpdateOrganizationUsersRequest = _interopRequireDefault(require("./model/UpdateOrganizationUsersRequest"));
var _UpdateOrganizationUsersRequestUsersInner = _interopRequireDefault(require("./model/UpdateOrganizationUsersRequestUsersInner"));
var _UpdateUserRequest = _interopRequireDefault(require("./model/UpdateUserRequest"));
var _User = _interopRequireDefault(require("./model/User"));
var _UserIdentity = _interopRequireDefault(require("./model/UserIdentity"));
var _UserIdentityResult = _interopRequireDefault(require("./model/UserIdentityResult"));
var _UserProfile = _interopRequireDefault(require("./model/UserProfile"));
var _UserProfileV = _interopRequireDefault(require("./model/UserProfileV2"));
var _UsersResponse = _interopRequireDefault(require("./model/UsersResponse"));
var _CallbacksApi = _interopRequireDefault(require("./api/CallbacksApi"));
var _ConnectedAppsApi = _interopRequireDefault(require("./api/ConnectedAppsApi"));
var _EnvironmentsApi = _interopRequireDefault(require("./api/EnvironmentsApi"));
var _FeatureFlagsApi = _interopRequireDefault(require("./api/FeatureFlagsApi"));
var _OAuthApi = _interopRequireDefault(require("./api/OAuthApi"));
var _OrganizationsApi = _interopRequireDefault(require("./api/OrganizationsApi"));
var _PermissionsApi = _interopRequireDefault(require("./api/PermissionsApi"));
var _RolesApi = _interopRequireDefault(require("./api/RolesApi"));
var _UsersApi = _interopRequireDefault(require("./api/UsersApi"));
function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { "default": obj }; }
