// @kinde-oss/kinde-nodejs-sdk/dist/model/AddOrganizationUsersRequestUsersInner.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;
var _ApiClient = _interopRequireDefault(require("../ApiClient"));
function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { "default": obj }; }
function _typeof(obj) { "@babel/helpers - typeof"; return _typeof = "function" == typeof Symbol && "symbol" == typeof Symbol.iterator ? function (obj) { return typeof obj; } : function (obj) { return obj && "function" == typeof Symbol && obj.constructor === Symbol && obj !== Symbol.prototype ? "symbol" : typeof obj; }, _typeof(obj); }
function _classCallCheck(instance, Constructor) { if (!(instance instanceof Constructor)) { throw new TypeError("Cannot call a class as a function"); } }
function _defineProperties(target, props) { for (var i = 0; i < props.length; i++) { var descriptor = props[i]; descriptor.enumerable = descriptor.enumerable || false; descriptor.configurable = true; if ("value" in descriptor) descriptor.writable = true; Object.defineProperty(target, _toPropertyKey(descriptor.key), descriptor); } }
function _createClass(Constructor, protoProps, staticProps) { if (protoProps) _defineProperties(Constructor.prototype, protoProps); if (staticProps) _defineProperties(Constructor, staticProps); Object.defineProperty(Constructor, "prototype", { writable: false }); return Constructor; }
function _toPropertyKey(arg) { var key = _toPrimitive(arg, "string"); return _typeof(key) === "symbol" ? key : String(key); }
function _toPrimitive(input, hint) { if (_typeof(input) !== "object" || input === null) return input; var prim = input[Symbol.toPrimitive]; if (prim !== undefined) { var res = prim.call(input, hint || "default"); if (_typeof(res) !== "object") return res; throw new TypeError("@@toPrimitive must return a primitive value."); } return (hint === "string" ? String : Number)(input); } /**
                                                                                                                                                                                                                                                                                                                                                                                               * Kinde Management API
                                                                                                                                                                                                                                                                                                                                                                                               * Provides endpoints to manage your Kinde Businesses
                                                                                                                                                                                                                                                                                                                                                                                               *
                                                                                                                                                                                                                                                                                                                                                                                               * The version of the OpenAPI document: 1
                                                                                                                                                                                                                                                                                                                                                                                               * Contact: support@kinde.com
                                                                                                                                                                                                                                                                                                                                                                                               *
                                                                                                                                                                                                                                                                                                                                                                                               * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
                                                                                                                                                                                                                                                                                                                                                                                               * https://openapi-generator.tech
                                                                                                                                                                                                                                                                                                                                                                                               * Do not edit the class manually.
                                                                                                                                                                                                                                                                                                                                                                                               *
                                                                                                                                                                                                                                                                                                                                                                                               */
/**
 * The AddOrganizationUsersRequestUsersInner model module.
 * @module model/AddOrganizationUsersRequestUsersInner
 * @version 1
 */
var AddOrganizationUsersRequestUsersInner = /*#__PURE__*/function () {
  /**
   * Constructs a new <code>AddOrganizationUsersRequestUsersInner</code>.
   * @alias module:model/AddOrganizationUsersRequestUsersInner
   */
  function AddOrganizationUsersRequestUsersInner() {
    _classCallCheck(this, AddOrganizationUsersRequestUsersInner);
    AddOrganizationUsersRequestUsersInner.initialize(this);
  }

  /**
   * Initializes the fields of this object.
   * This method is used by the constructors of any subclasses, in order to implement multiple inheritance (mix-ins).
   * Only for internal use.
   */
  _createClass(AddOrganizationUsersRequestUsersInner, null, [{
    key: "initialize",
    value: function initialize(obj) {}

    /**
     * Constructs a <code>AddOrganizationUsersRequestUsersInner</code> from a plain JavaScript object, optionally creating a new instance.
     * Copies all relevant properties from <code>data</code> to <code>obj</code> if supplied or a new instance if not.
     * @param {Object} data The plain JavaScript object bearing properties of interest.
     * @param {module:model/AddOrganizationUsersRequestUsersInner} obj Optional instance to populate.
     * @return {module:model/AddOrganizationUsersRequestUsersInner} The populated <code>AddOrganizationUsersRequestUsersInner</code> instance.
     */
  }, {
    key: "constructFromObject",
    value: function constructFromObject(data, obj) {
      if (data) {
        obj = obj || new AddOrganizationUsersRequestUsersInner();
        if (data.hasOwnProperty('id')) {
          obj['id'] = _ApiClient["default"].convertToType(data['id'], 'String');
        }
        if (data.hasOwnProperty('roles')) {
          obj['roles'] = _ApiClient["default"].convertToType(data['roles'], ['String']);
        }
        if (data.hasOwnProperty('permissions')) {
          obj['permissions'] = _ApiClient["default"].convertToType(data['permissions'], ['String']);
        }
      }
      return obj;
    }

    /**
     * Validates the JSON data with respect to <code>AddOrganizationUsersRequestUsersInner</code>.
     * @param {Object} data The plain JavaScript object bearing properties of interest.
     * @return {boolean} to indicate whether the JSON data is valid with respect to <code>AddOrganizationUsersRequestUsersInner</code>.
     */
  }, {
    key: "validateJSON",
    value: function validateJSON(data) {
      // ensure the json data is a string
      if (data['id'] && !(typeof data['id'] === 'string' || data['id'] instanceof String)) {
        throw new Error("Expected the field `id` to be a primitive type in the JSON string but got " + data['id']);
      }
      // ensure the json data is an array
      if (!Array.isArray(data['roles'])) {
        throw new Error("Expected the field `roles` to be an array in the JSON data but got " + data['roles']);
      }
      // ensure the json data is an array
      if (!Array.isArray(data['permissions'])) {
        throw new Error("Expected the field `permissions` to be an array in the JSON data but got " + data['permissions']);
      }
      return true;
    }
  }]);
  return AddOrganizationUsersRequestUsersInner;
}();
/**
 * The users id.
 * @member {String} id
 */
AddOrganizationUsersRequestUsersInner.prototype['id'] = undefined;

/**
 * Role keys to assign to the user.
 * @member {Array.<String>} roles
 */
AddOrganizationUsersRequestUsersInner.prototype['roles'] = undefined;

/**
 * Permission keys to assign to the user.
 * @member {Array.<String>} permissions
 */
AddOrganizationUsersRequestUsersInner.prototype['permissions'] = undefined;
var _default = AddOrganizationUsersRequestUsersInner;
exports["default"] = _default;
