// @kinde-oss/kinde-nodejs-sdk/dist/model/ConnectedAppsAccessToken.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;
var _ApiClient = _interopRequireDefault(require("../ApiClient"));
function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { "default": obj }; }
function _typeof(obj) { "@babel/helpers - typeof"; return _typeof = "function" == typeof Symbol && "symbol" == typeof Symbol.iterator ? function (obj) { return typeof obj; } : function (obj) { return obj && "function" == typeof Symbol && obj.constructor === Symbol && obj !== Symbol.prototype ? "symbol" : typeof obj; }, _typeof(obj); }
function _classCallCheck(instance, Constructor) { if (!(instance instanceof Constructor)) { throw new TypeError("Cannot call a class as a function"); } }
function _defineProperties(target, props) { for (var i = 0; i < props.length; i++) { var descriptor = props[i]; descriptor.enumerable = descriptor.enumerable || false; descriptor.configurable = true; if ("value" in descriptor) descriptor.writable = true; Object.defineProperty(target, _toPropertyKey(descriptor.key), descriptor); } }
function _createClass(Constructor, protoProps, staticProps) { if (protoProps) _defineProperties(Constructor.prototype, protoProps); if (staticProps) _defineProperties(Constructor, staticProps); Object.defineProperty(Constructor, "prototype", { writable: false }); return Constructor; }
function _toPropertyKey(arg) { var key = _toPrimitive(arg, "string"); return _typeof(key) === "symbol" ? key : String(key); }
function _toPrimitive(input, hint) { if (_typeof(input) !== "object" || input === null) return input; var prim = input[Symbol.toPrimitive]; if (prim !== undefined) { var res = prim.call(input, hint || "default"); if (_typeof(res) !== "object") return res; throw new TypeError("@@toPrimitive must return a primitive value."); } return (hint === "string" ? String : Number)(input); } /**
                                                                                                                                                                                                                                                                                                                                                                                               * Kinde Management API
                                                                                                                                                                                                                                                                                                                                                                                               * Provides endpoints to manage your Kinde Businesses
                                                                                                                                                                                                                                                                                                                                                                                               *
                                                                                                                                                                                                                                                                                                                                                                                               * The version of the OpenAPI document: 1
                                                                                                                                                                                                                                                                                                                                                                                               * Contact: support@kinde.com
                                                                                                                                                                                                                                                                                                                                                                                               *
                                                                                                                                                                                                                                                                                                                                                                                               * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
                                                                                                                                                                                                                                                                                                                                                                                               * https://openapi-generator.tech
                                                                                                                                                                                                                                                                                                                                                                                               * Do not edit the class manually.
                                                                                                                                                                                                                                                                                                                                                                                               *
                                                                                                                                                                                                                                                                                                                                                                                               */
/**
 * The ConnectedAppsAccessToken model module.
 * @module model/ConnectedAppsAccessToken
 * @version 1
 */
var ConnectedAppsAccessToken = /*#__PURE__*/function () {
  /**
   * Constructs a new <code>ConnectedAppsAccessToken</code>.
   * @alias module:model/ConnectedAppsAccessToken
   */
  function ConnectedAppsAccessToken() {
    _classCallCheck(this, ConnectedAppsAccessToken);
    ConnectedAppsAccessToken.initialize(this);
  }

  /**
   * Initializes the fields of this object.
   * This method is used by the constructors of any subclasses, in order to implement multiple inheritance (mix-ins).
   * Only for internal use.
   */
  _createClass(ConnectedAppsAccessToken, null, [{
    key: "initialize",
    value: function initialize(obj) {}

    /**
     * Constructs a <code>ConnectedAppsAccessToken</code> from a plain JavaScript object, optionally creating a new instance.
     * Copies all relevant properties from <code>data</code> to <code>obj</code> if supplied or a new instance if not.
     * @param {Object} data The plain JavaScript object bearing properties of interest.
     * @param {module:model/ConnectedAppsAccessToken} obj Optional instance to populate.
     * @return {module:model/ConnectedAppsAccessToken} The populated <code>ConnectedAppsAccessToken</code> instance.
     */
  }, {
    key: "constructFromObject",
    value: function constructFromObject(data, obj) {
      if (data) {
        obj = obj || new ConnectedAppsAccessToken();
        if (data.hasOwnProperty('access_token')) {
          obj['access_token'] = _ApiClient["default"].convertToType(data['access_token'], 'String');
        }
        if (data.hasOwnProperty('access_token_expiry')) {
          obj['access_token_expiry'] = _ApiClient["default"].convertToType(data['access_token_expiry'], 'String');
        }
      }
      return obj;
    }

    /**
     * Validates the JSON data with respect to <code>ConnectedAppsAccessToken</code>.
     * @param {Object} data The plain JavaScript object bearing properties of interest.
     * @return {boolean} to indicate whether the JSON data is valid with respect to <code>ConnectedAppsAccessToken</code>.
     */
  }, {
    key: "validateJSON",
    value: function validateJSON(data) {
      // ensure the json data is a string
      if (data['access_token'] && !(typeof data['access_token'] === 'string' || data['access_token'] instanceof String)) {
        throw new Error("Expected the field `access_token` to be a primitive type in the JSON string but got " + data['access_token']);
      }
      // ensure the json data is a string
      if (data['access_token_expiry'] && !(typeof data['access_token_expiry'] === 'string' || data['access_token_expiry'] instanceof String)) {
        throw new Error("Expected the field `access_token_expiry` to be a primitive type in the JSON string but got " + data['access_token_expiry']);
      }
      return true;
    }
  }]);
  return ConnectedAppsAccessToken;
}();
/**
 * The access token to access a third-party provider.
 * @member {String} access_token
 */
ConnectedAppsAccessToken.prototype['access_token'] = undefined;

/**
 * The date and time that the access token expires.
 * @member {String} access_token_expiry
 */
ConnectedAppsAccessToken.prototype['access_token_expiry'] = undefined;
var _default = ConnectedAppsAccessToken;
exports["default"] = _default;
