// @kinde-oss/kinde-nodejs-sdk/dist/model/UserProfile.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;
var _ApiClient = _interopRequireDefault(require("../ApiClient"));
function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { "default": obj }; }
function _typeof(obj) { "@babel/helpers - typeof"; return _typeof = "function" == typeof Symbol && "symbol" == typeof Symbol.iterator ? function (obj) { return typeof obj; } : function (obj) { return obj && "function" == typeof Symbol && obj.constructor === Symbol && obj !== Symbol.prototype ? "symbol" : typeof obj; }, _typeof(obj); }
function _classCallCheck(instance, Constructor) { if (!(instance instanceof Constructor)) { throw new TypeError("Cannot call a class as a function"); } }
function _defineProperties(target, props) { for (var i = 0; i < props.length; i++) { var descriptor = props[i]; descriptor.enumerable = descriptor.enumerable || false; descriptor.configurable = true; if ("value" in descriptor) descriptor.writable = true; Object.defineProperty(target, _toPropertyKey(descriptor.key), descriptor); } }
function _createClass(Constructor, protoProps, staticProps) { if (protoProps) _defineProperties(Constructor.prototype, protoProps); if (staticProps) _defineProperties(Constructor, staticProps); Object.defineProperty(Constructor, "prototype", { writable: false }); return Constructor; }
function _toPropertyKey(arg) { var key = _toPrimitive(arg, "string"); return _typeof(key) === "symbol" ? key : String(key); }
function _toPrimitive(input, hint) { if (_typeof(input) !== "object" || input === null) return input; var prim = input[Symbol.toPrimitive]; if (prim !== undefined) { var res = prim.call(input, hint || "default"); if (_typeof(res) !== "object") return res; throw new TypeError("@@toPrimitive must return a primitive value."); } return (hint === "string" ? String : Number)(input); } /**
                                                                                                                                                                                                                                                                                                                                                                                               * Kinde Management API
                                                                                                                                                                                                                                                                                                                                                                                               * Provides endpoints to manage your Kinde Businesses
                                                                                                                                                                                                                                                                                                                                                                                               *
                                                                                                                                                                                                                                                                                                                                                                                               * The version of the OpenAPI document: 1
                                                                                                                                                                                                                                                                                                                                                                                               * Contact: support@kinde.com
                                                                                                                                                                                                                                                                                                                                                                                               *
                                                                                                                                                                                                                                                                                                                                                                                               * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
                                                                                                                                                                                                                                                                                                                                                                                               * https://openapi-generator.tech
                                                                                                                                                                                                                                                                                                                                                                                               * Do not edit the class manually.
                                                                                                                                                                                                                                                                                                                                                                                               *
                                                                                                                                                                                                                                                                                                                                                                                               */
/**
 * The UserProfile model module.
 * @module model/UserProfile
 * @version 1
 */
var UserProfile = /*#__PURE__*/function () {
  /**
   * Constructs a new <code>UserProfile</code>.
   * @alias module:model/UserProfile
   */
  function UserProfile() {
    _classCallCheck(this, UserProfile);
    UserProfile.initialize(this);
  }

  /**
   * Initializes the fields of this object.
   * This method is used by the constructors of any subclasses, in order to implement multiple inheritance (mix-ins).
   * Only for internal use.
   */
  _createClass(UserProfile, null, [{
    key: "initialize",
    value: function initialize(obj) {}

    /**
     * Constructs a <code>UserProfile</code> from a plain JavaScript object, optionally creating a new instance.
     * Copies all relevant properties from <code>data</code> to <code>obj</code> if supplied or a new instance if not.
     * @param {Object} data The plain JavaScript object bearing properties of interest.
     * @param {module:model/UserProfile} obj Optional instance to populate.
     * @return {module:model/UserProfile} The populated <code>UserProfile</code> instance.
     */
  }, {
    key: "constructFromObject",
    value: function constructFromObject(data, obj) {
      if (data) {
        obj = obj || new UserProfile();
        if (data.hasOwnProperty('id')) {
          obj['id'] = _ApiClient["default"].convertToType(data['id'], 'String');
        }
        if (data.hasOwnProperty('preferred_email')) {
          obj['preferred_email'] = _ApiClient["default"].convertToType(data['preferred_email'], 'String');
        }
        if (data.hasOwnProperty('provided_id')) {
          obj['provided_id'] = _ApiClient["default"].convertToType(data['provided_id'], 'String');
        }
        if (data.hasOwnProperty('last_name')) {
          obj['last_name'] = _ApiClient["default"].convertToType(data['last_name'], 'String');
        }
        if (data.hasOwnProperty('first_name')) {
          obj['first_name'] = _ApiClient["default"].convertToType(data['first_name'], 'String');
        }
        if (data.hasOwnProperty('picture')) {
          obj['picture'] = _ApiClient["default"].convertToType(data['picture'], 'String');
        }
      }
      return obj;
    }

    /**
     * Validates the JSON data with respect to <code>UserProfile</code>.
     * @param {Object} data The plain JavaScript object bearing properties of interest.
     * @return {boolean} to indicate whether the JSON data is valid with respect to <code>UserProfile</code>.
     */
  }, {
    key: "validateJSON",
    value: function validateJSON(data) {
      // ensure the json data is a string
      if (data['id'] && !(typeof data['id'] === 'string' || data['id'] instanceof String)) {
        throw new Error("Expected the field `id` to be a primitive type in the JSON string but got " + data['id']);
      }
      // ensure the json data is a string
      if (data['preferred_email'] && !(typeof data['preferred_email'] === 'string' || data['preferred_email'] instanceof String)) {
        throw new Error("Expected the field `preferred_email` to be a primitive type in the JSON string but got " + data['preferred_email']);
      }
      // ensure the json data is a string
      if (data['provided_id'] && !(typeof data['provided_id'] === 'string' || data['provided_id'] instanceof String)) {
        throw new Error("Expected the field `provided_id` to be a primitive type in the JSON string but got " + data['provided_id']);
      }
      // ensure the json data is a string
      if (data['last_name'] && !(typeof data['last_name'] === 'string' || data['last_name'] instanceof String)) {
        throw new Error("Expected the field `last_name` to be a primitive type in the JSON string but got " + data['last_name']);
      }
      // ensure the json data is a string
      if (data['first_name'] && !(typeof data['first_name'] === 'string' || data['first_name'] instanceof String)) {
        throw new Error("Expected the field `first_name` to be a primitive type in the JSON string but got " + data['first_name']);
      }
      // ensure the json data is a string
      if (data['picture'] && !(typeof data['picture'] === 'string' || data['picture'] instanceof String)) {
        throw new Error("Expected the field `picture` to be a primitive type in the JSON string but got " + data['picture']);
      }
      return true;
    }
  }]);
  return UserProfile;
}();
/**
 * Unique id of the user in Kinde.
 * @member {String} id
 */
UserProfile.prototype['id'] = undefined;

/**
 * Default email address of the user in Kinde.
 * @member {String} preferred_email
 */
UserProfile.prototype['preferred_email'] = undefined;

/**
 * Value of the user's id in a third-party system when the user is imported into Kinde.
 * @member {String} provided_id
 */
UserProfile.prototype['provided_id'] = undefined;

/**
 * User's last name.
 * @member {String} last_name
 */
UserProfile.prototype['last_name'] = undefined;

/**
 * User's first name.
 * @member {String} first_name
 */
UserProfile.prototype['first_name'] = undefined;

/**
 * URL that point's to the user's picture or avatar
 * @member {String} picture
 */
UserProfile.prototype['picture'] = undefined;
var _default = UserProfile;
exports["default"] = _default;
