// @kinde-oss/kinde-nodejs-sdk/dist/model/UserProfileV2.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;
var _ApiClient = _interopRequireDefault(require("../ApiClient"));
function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { "default": obj }; }
function _typeof(obj) { "@babel/helpers - typeof"; return _typeof = "function" == typeof Symbol && "symbol" == typeof Symbol.iterator ? function (obj) { return typeof obj; } : function (obj) { return obj && "function" == typeof Symbol && obj.constructor === Symbol && obj !== Symbol.prototype ? "symbol" : typeof obj; }, _typeof(obj); }
function _classCallCheck(instance, Constructor) { if (!(instance instanceof Constructor)) { throw new TypeError("Cannot call a class as a function"); } }
function _defineProperties(target, props) { for (var i = 0; i < props.length; i++) { var descriptor = props[i]; descriptor.enumerable = descriptor.enumerable || false; descriptor.configurable = true; if ("value" in descriptor) descriptor.writable = true; Object.defineProperty(target, _toPropertyKey(descriptor.key), descriptor); } }
function _createClass(Constructor, protoProps, staticProps) { if (protoProps) _defineProperties(Constructor.prototype, protoProps); if (staticProps) _defineProperties(Constructor, staticProps); Object.defineProperty(Constructor, "prototype", { writable: false }); return Constructor; }
function _toPropertyKey(arg) { var key = _toPrimitive(arg, "string"); return _typeof(key) === "symbol" ? key : String(key); }
function _toPrimitive(input, hint) { if (_typeof(input) !== "object" || input === null) return input; var prim = input[Symbol.toPrimitive]; if (prim !== undefined) { var res = prim.call(input, hint || "default"); if (_typeof(res) !== "object") return res; throw new TypeError("@@toPrimitive must return a primitive value."); } return (hint === "string" ? String : Number)(input); } /**
                                                                                                                                                                                                                                                                                                                                                                                               * Kinde Management API
                                                                                                                                                                                                                                                                                                                                                                                               * Provides endpoints to manage your Kinde Businesses
                                                                                                                                                                                                                                                                                                                                                                                               *
                                                                                                                                                                                                                                                                                                                                                                                               * The version of the OpenAPI document: 1
                                                                                                                                                                                                                                                                                                                                                                                               * Contact: support@kinde.com
                                                                                                                                                                                                                                                                                                                                                                                               *
                                                                                                                                                                                                                                                                                                                                                                                               * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
                                                                                                                                                                                                                                                                                                                                                                                               * https://openapi-generator.tech
                                                                                                                                                                                                                                                                                                                                                                                               * Do not edit the class manually.
                                                                                                                                                                                                                                                                                                                                                                                               *
                                                                                                                                                                                                                                                                                                                                                                                               */
/**
 * The UserProfileV2 model module.
 * @module model/UserProfileV2
 * @version 1
 */
var UserProfileV2 = /*#__PURE__*/function () {
  /**
   * Constructs a new <code>UserProfileV2</code>.
   * @alias module:model/UserProfileV2
   */
  function UserProfileV2() {
    _classCallCheck(this, UserProfileV2);
    UserProfileV2.initialize(this);
  }

  /**
   * Initializes the fields of this object.
   * This method is used by the constructors of any subclasses, in order to implement multiple inheritance (mix-ins).
   * Only for internal use.
   */
  _createClass(UserProfileV2, null, [{
    key: "initialize",
    value: function initialize(obj) {}

    /**
     * Constructs a <code>UserProfileV2</code> from a plain JavaScript object, optionally creating a new instance.
     * Copies all relevant properties from <code>data</code> to <code>obj</code> if supplied or a new instance if not.
     * @param {Object} data The plain JavaScript object bearing properties of interest.
     * @param {module:model/UserProfileV2} obj Optional instance to populate.
     * @return {module:model/UserProfileV2} The populated <code>UserProfileV2</code> instance.
     */
  }, {
    key: "constructFromObject",
    value: function constructFromObject(data, obj) {
      if (data) {
        obj = obj || new UserProfileV2();
        if (data.hasOwnProperty('id')) {
          obj['id'] = _ApiClient["default"].convertToType(data['id'], 'String');
        }
        if (data.hasOwnProperty('sub')) {
          obj['sub'] = _ApiClient["default"].convertToType(data['sub'], 'String');
        }
        if (data.hasOwnProperty('provided_id')) {
          obj['provided_id'] = _ApiClient["default"].convertToType(data['provided_id'], 'String');
        }
        if (data.hasOwnProperty('name')) {
          obj['name'] = _ApiClient["default"].convertToType(data['name'], 'String');
        }
        if (data.hasOwnProperty('given_name')) {
          obj['given_name'] = _ApiClient["default"].convertToType(data['given_name'], 'String');
        }
        if (data.hasOwnProperty('family_name')) {
          obj['family_name'] = _ApiClient["default"].convertToType(data['family_name'], 'String');
        }
        if (data.hasOwnProperty('updated_at')) {
          obj['updated_at'] = _ApiClient["default"].convertToType(data['updated_at'], 'Number');
        }
        if (data.hasOwnProperty('email')) {
          obj['email'] = _ApiClient["default"].convertToType(data['email'], 'String');
        }
        if (data.hasOwnProperty('picture')) {
          obj['picture'] = _ApiClient["default"].convertToType(data['picture'], 'String');
        }
      }
      return obj;
    }

    /**
     * Validates the JSON data with respect to <code>UserProfileV2</code>.
     * @param {Object} data The plain JavaScript object bearing properties of interest.
     * @return {boolean} to indicate whether the JSON data is valid with respect to <code>UserProfileV2</code>.
     */
  }, {
    key: "validateJSON",
    value: function validateJSON(data) {
      // ensure the json data is a string
      if (data['id'] && !(typeof data['id'] === 'string' || data['id'] instanceof String)) {
        throw new Error("Expected the field `id` to be a primitive type in the JSON string but got " + data['id']);
      }
      // ensure the json data is a string
      if (data['sub'] && !(typeof data['sub'] === 'string' || data['sub'] instanceof String)) {
        throw new Error("Expected the field `sub` to be a primitive type in the JSON string but got " + data['sub']);
      }
      // ensure the json data is a string
      if (data['provided_id'] && !(typeof data['provided_id'] === 'string' || data['provided_id'] instanceof String)) {
        throw new Error("Expected the field `provided_id` to be a primitive type in the JSON string but got " + data['provided_id']);
      }
      // ensure the json data is a string
      if (data['name'] && !(typeof data['name'] === 'string' || data['name'] instanceof String)) {
        throw new Error("Expected the field `name` to be a primitive type in the JSON string but got " + data['name']);
      }
      // ensure the json data is a string
      if (data['given_name'] && !(typeof data['given_name'] === 'string' || data['given_name'] instanceof String)) {
        throw new Error("Expected the field `given_name` to be a primitive type in the JSON string but got " + data['given_name']);
      }
      // ensure the json data is a string
      if (data['family_name'] && !(typeof data['family_name'] === 'string' || data['family_name'] instanceof String)) {
        throw new Error("Expected the field `family_name` to be a primitive type in the JSON string but got " + data['family_name']);
      }
      // ensure the json data is a string
      if (data['email'] && !(typeof data['email'] === 'string' || data['email'] instanceof String)) {
        throw new Error("Expected the field `email` to be a primitive type in the JSON string but got " + data['email']);
      }
      // ensure the json data is a string
      if (data['picture'] && !(typeof data['picture'] === 'string' || data['picture'] instanceof String)) {
        throw new Error("Expected the field `picture` to be a primitive type in the JSON string but got " + data['picture']);
      }
      return true;
    }
  }]);
  return UserProfileV2;
}();
/**
 * Unique id of the user in Kinde (deprecated).
 * @member {String} id
 */
UserProfileV2.prototype['id'] = undefined;

/**
 * Unique id of the user in Kinde.
 * @member {String} sub
 */
UserProfileV2.prototype['sub'] = undefined;

/**
 * Value of the user's id in a third-party system when the user is imported into Kinde.
 * @member {String} provided_id
 */
UserProfileV2.prototype['provided_id'] = undefined;

/**
 * Users's first and last name separated by a space.
 * @member {String} name
 */
UserProfileV2.prototype['name'] = undefined;

/**
 * User's first name.
 * @member {String} given_name
 */
UserProfileV2.prototype['given_name'] = undefined;

/**
 * User's last name.
 * @member {String} family_name
 */
UserProfileV2.prototype['family_name'] = undefined;

/**
 * Date the user was last updated at (In Unix time).
 * @member {Number} updated_at
 */
UserProfileV2.prototype['updated_at'] = undefined;

/**
 * User's email address if available.
 * @member {String} email
 */
UserProfileV2.prototype['email'] = undefined;

/**
 * URL that point's to the user's picture or avatar
 * @member {String} picture
 */
UserProfileV2.prototype['picture'] = undefined;
var _default = UserProfileV2;
exports["default"] = _default;
