// @kinde-oss/kinde-nodejs-sdk/dist/api/ConnectedAppsApi.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;
var _ApiClient = _interopRequireDefault(require("../ApiClient"));
var _ConnectedAppsAccessToken = _interopRequireDefault(require("../model/ConnectedAppsAccessToken"));
var _ConnectedAppsAuthUrl = _interopRequireDefault(require("../model/ConnectedAppsAuthUrl"));
var _ErrorResponse = _interopRequireDefault(require("../model/ErrorResponse"));
var _SuccessResponse = _interopRequireDefault(require("../model/SuccessResponse"));
function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { "default": obj }; }
function _typeof(obj) { "@babel/helpers - typeof"; return _typeof = "function" == typeof Symbol && "symbol" == typeof Symbol.iterator ? function (obj) { return typeof obj; } : function (obj) { return obj && "function" == typeof Symbol && obj.constructor === Symbol && obj !== Symbol.prototype ? "symbol" : typeof obj; }, _typeof(obj); }
function _classCallCheck(instance, Constructor) { if (!(instance instanceof Constructor)) { throw new TypeError("Cannot call a class as a function"); } }
function _defineProperties(target, props) { for (var i = 0; i < props.length; i++) { var descriptor = props[i]; descriptor.enumerable = descriptor.enumerable || false; descriptor.configurable = true; if ("value" in descriptor) descriptor.writable = true; Object.defineProperty(target, _toPropertyKey(descriptor.key), descriptor); } }
function _createClass(Constructor, protoProps, staticProps) { if (protoProps) _defineProperties(Constructor.prototype, protoProps); if (staticProps) _defineProperties(Constructor, staticProps); Object.defineProperty(Constructor, "prototype", { writable: false }); return Constructor; }
function _toPropertyKey(arg) { var key = _toPrimitive(arg, "string"); return _typeof(key) === "symbol" ? key : String(key); }
function _toPrimitive(input, hint) { if (_typeof(input) !== "object" || input === null) return input; var prim = input[Symbol.toPrimitive]; if (prim !== undefined) { var res = prim.call(input, hint || "default"); if (_typeof(res) !== "object") return res; throw new TypeError("@@toPrimitive must return a primitive value."); } return (hint === "string" ? String : Number)(input); } /**
                                                                                                                                                                                                                                                                                                                                                                                               * Kinde Management API
                                                                                                                                                                                                                                                                                                                                                                                               * Provides endpoints to manage your Kinde Businesses
                                                                                                                                                                                                                                                                                                                                                                                               *
                                                                                                                                                                                                                                                                                                                                                                                               * The version of the OpenAPI document: 1
                                                                                                                                                                                                                                                                                                                                                                                               * Contact: support@kinde.com
                                                                                                                                                                                                                                                                                                                                                                                               *
                                                                                                                                                                                                                                                                                                                                                                                               * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
                                                                                                                                                                                                                                                                                                                                                                                               * https://openapi-generator.tech
                                                                                                                                                                                                                                                                                                                                                                                               * Do not edit the class manually.
                                                                                                                                                                                                                                                                                                                                                                                               *
                                                                                                                                                                                                                                                                                                                                                                                               */
/**
* ConnectedApps service.
* @module api/ConnectedAppsApi
* @version 1
*/
var ConnectedAppsApi = /*#__PURE__*/function () {
  /**
  * Constructs a new ConnectedAppsApi. 
  * @alias module:api/ConnectedAppsApi
  * @class
  * @param {module:ApiClient} [apiClient] Optional API client implementation to use,
  * default to {@link module:ApiClient#instance} if unspecified.
  */
  function ConnectedAppsApi(apiClient) {
    _classCallCheck(this, ConnectedAppsApi);
    this.apiClient = apiClient || _ApiClient["default"].instance;
  }

  /**
   * Callback function to receive the result of the getConnectedAppAuthUrl operation.
   * @callback module:api/ConnectedAppsApi~getConnectedAppAuthUrlCallback
   * @param {String} error Error message, if any.
   * @param {module:model/ConnectedAppsAuthUrl} data The data returned by the service call.
   * @param {String} response The complete HTTP response.
   */

  /**
   * Get Connected App URL
   * Get a URL that authenticates and authorizes a user to a third-party connected app.
   * @param {String} keyCodeRef The unique key code reference of the connected app to authenticate against.
   * @param {String} userId The id of the user that needs to authenticate to the third-party connected app.
   * @param {module:api/ConnectedAppsApi~getConnectedAppAuthUrlCallback} callback The callback function, accepting three arguments: error, data, response
   * data is of type: {@link module:model/ConnectedAppsAuthUrl}
   */
  _createClass(ConnectedAppsApi, [{
    key: "getConnectedAppAuthUrl",
    value: function getConnectedAppAuthUrl(keyCodeRef, userId, callback) {
      var postBody = null;
      // verify the required parameter 'keyCodeRef' is set
      if (keyCodeRef === undefined || keyCodeRef === null) {
        throw new Error("Missing the required parameter 'keyCodeRef' when calling getConnectedAppAuthUrl");
      }
      // verify the required parameter 'userId' is set
      if (userId === undefined || userId === null) {
        throw new Error("Missing the required parameter 'userId' when calling getConnectedAppAuthUrl");
      }
      var pathParams = {};
      var queryParams = {
        'key_code_ref': keyCodeRef,
        'user_id': userId
      };
      var headerParams = {};
      var formParams = {};
      var authNames = ['kindeBearerAuth'];
      var contentTypes = [];
      var accepts = ['application/json', 'application/json; charset=utf-8'];
      var returnType = _ConnectedAppsAuthUrl["default"];
      return this.apiClient.callApi('/api/v1/connected_apps/auth_url', 'GET', pathParams, queryParams, headerParams, formParams, postBody, authNames, contentTypes, accepts, returnType, null, callback);
    }

    /**
     * Callback function to receive the result of the getConnectedAppToken operation.
     * @callback module:api/ConnectedAppsApi~getConnectedAppTokenCallback
     * @param {String} error Error message, if any.
     * @param {module:model/ConnectedAppsAccessToken} data The data returned by the service call.
     * @param {String} response The complete HTTP response.
     */

    /**
     * Get Connected App Token
     * Get an access token that can be used to call the third-party provider linked to the connected app.
     * @param {String} sessionId The unique sesssion id reprensenting the login session of a user.
     * @param {module:api/ConnectedAppsApi~getConnectedAppTokenCallback} callback The callback function, accepting three arguments: error, data, response
     * data is of type: {@link module:model/ConnectedAppsAccessToken}
     */
  }, {
    key: "getConnectedAppToken",
    value: function getConnectedAppToken(sessionId, callback) {
      var postBody = null;
      // verify the required parameter 'sessionId' is set
      if (sessionId === undefined || sessionId === null) {
        throw new Error("Missing the required parameter 'sessionId' when calling getConnectedAppToken");
      }
      var pathParams = {};
      var queryParams = {
        'session_id': sessionId
      };
      var headerParams = {};
      var formParams = {};
      var authNames = ['kindeBearerAuth'];
      var contentTypes = [];
      var accepts = ['application/json', 'application/json; charset=utf-8'];
      var returnType = _ConnectedAppsAccessToken["default"];
      return this.apiClient.callApi('/api/v1/connected_apps/token', 'GET', pathParams, queryParams, headerParams, formParams, postBody, authNames, contentTypes, accepts, returnType, null, callback);
    }

    /**
     * Callback function to receive the result of the revokeConnectedAppToken operation.
     * @callback module:api/ConnectedAppsApi~revokeConnectedAppTokenCallback
     * @param {String} error Error message, if any.
     * @param {module:model/SuccessResponse} data The data returned by the service call.
     * @param {String} response The complete HTTP response.
     */

    /**
     * Revoke Connected App Token
     * Revoke the tokens linked to the connected app session.
     * @param {String} sessionId The unique sesssion id reprensenting the login session of a user.
     * @param {module:api/ConnectedAppsApi~revokeConnectedAppTokenCallback} callback The callback function, accepting three arguments: error, data, response
     * data is of type: {@link module:model/SuccessResponse}
     */
  }, {
    key: "revokeConnectedAppToken",
    value: function revokeConnectedAppToken(sessionId, callback) {
      var postBody = null;
      // verify the required parameter 'sessionId' is set
      if (sessionId === undefined || sessionId === null) {
        throw new Error("Missing the required parameter 'sessionId' when calling revokeConnectedAppToken");
      }
      var pathParams = {};
      var queryParams = {
        'session_id': sessionId
      };
      var headerParams = {};
      var formParams = {};
      var authNames = ['kindeBearerAuth'];
      var contentTypes = [];
      var accepts = ['application/json', 'application/json; charset=utf-8'];
      var returnType = _SuccessResponse["default"];
      return this.apiClient.callApi('/api/v1/connected_apps/revoke', 'POST', pathParams, queryParams, headerParams, formParams, postBody, authNames, contentTypes, accepts, returnType, null, callback);
    }
  }]);
  return ConnectedAppsApi;
}();
exports["default"] = ConnectedAppsApi;
