// @kinde-oss/kinde-nodejs-sdk/dist/api/EnvironmentsApi.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;
var _ApiClient = _interopRequireDefault(require("../ApiClient"));
var _ErrorResponse = _interopRequireDefault(require("../model/ErrorResponse"));
var _SuccessResponse = _interopRequireDefault(require("../model/SuccessResponse"));
function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { "default": obj }; }
function _typeof(obj) { "@babel/helpers - typeof"; return _typeof = "function" == typeof Symbol && "symbol" == typeof Symbol.iterator ? function (obj) { return typeof obj; } : function (obj) { return obj && "function" == typeof Symbol && obj.constructor === Symbol && obj !== Symbol.prototype ? "symbol" : typeof obj; }, _typeof(obj); }
function _classCallCheck(instance, Constructor) { if (!(instance instanceof Constructor)) { throw new TypeError("Cannot call a class as a function"); } }
function _defineProperties(target, props) { for (var i = 0; i < props.length; i++) { var descriptor = props[i]; descriptor.enumerable = descriptor.enumerable || false; descriptor.configurable = true; if ("value" in descriptor) descriptor.writable = true; Object.defineProperty(target, _toPropertyKey(descriptor.key), descriptor); } }
function _createClass(Constructor, protoProps, staticProps) { if (protoProps) _defineProperties(Constructor.prototype, protoProps); if (staticProps) _defineProperties(Constructor, staticProps); Object.defineProperty(Constructor, "prototype", { writable: false }); return Constructor; }
function _toPropertyKey(arg) { var key = _toPrimitive(arg, "string"); return _typeof(key) === "symbol" ? key : String(key); }
function _toPrimitive(input, hint) { if (_typeof(input) !== "object" || input === null) return input; var prim = input[Symbol.toPrimitive]; if (prim !== undefined) { var res = prim.call(input, hint || "default"); if (_typeof(res) !== "object") return res; throw new TypeError("@@toPrimitive must return a primitive value."); } return (hint === "string" ? String : Number)(input); } /**
                                                                                                                                                                                                                                                                                                                                                                                               * Kinde Management API
                                                                                                                                                                                                                                                                                                                                                                                               * Provides endpoints to manage your Kinde Businesses
                                                                                                                                                                                                                                                                                                                                                                                               *
                                                                                                                                                                                                                                                                                                                                                                                               * The version of the OpenAPI document: 1
                                                                                                                                                                                                                                                                                                                                                                                               * Contact: support@kinde.com
                                                                                                                                                                                                                                                                                                                                                                                               *
                                                                                                                                                                                                                                                                                                                                                                                               * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
                                                                                                                                                                                                                                                                                                                                                                                               * https://openapi-generator.tech
                                                                                                                                                                                                                                                                                                                                                                                               * Do not edit the class manually.
                                                                                                                                                                                                                                                                                                                                                                                               *
                                                                                                                                                                                                                                                                                                                                                                                               */
/**
* Environments service.
* @module api/EnvironmentsApi
* @version 1
*/
var EnvironmentsApi = /*#__PURE__*/function () {
  /**
  * Constructs a new EnvironmentsApi. 
  * @alias module:api/EnvironmentsApi
  * @class
  * @param {module:ApiClient} [apiClient] Optional API client implementation to use,
  * default to {@link module:ApiClient#instance} if unspecified.
  */
  function EnvironmentsApi(apiClient) {
    _classCallCheck(this, EnvironmentsApi);
    this.apiClient = apiClient || _ApiClient["default"].instance;
  }

  /**
   * Callback function to receive the result of the deleteEnvironementFeatureFlagOverride operation.
   * @callback module:api/EnvironmentsApi~deleteEnvironementFeatureFlagOverrideCallback
   * @param {String} error Error message, if any.
   * @param {module:model/SuccessResponse} data The data returned by the service call.
   * @param {String} response The complete HTTP response.
   */

  /**
   * Delete environment feature flag override
   * Delete environment feature flag override.
   * @param {String} featureFlagKey The identifier for the feature flag.
   * @param {module:api/EnvironmentsApi~deleteEnvironementFeatureFlagOverrideCallback} callback The callback function, accepting three arguments: error, data, response
   * data is of type: {@link module:model/SuccessResponse}
   */
  _createClass(EnvironmentsApi, [{
    key: "deleteEnvironementFeatureFlagOverride",
    value: function deleteEnvironementFeatureFlagOverride(featureFlagKey, callback) {
      var postBody = null;
      // verify the required parameter 'featureFlagKey' is set
      if (featureFlagKey === undefined || featureFlagKey === null) {
        throw new Error("Missing the required parameter 'featureFlagKey' when calling deleteEnvironementFeatureFlagOverride");
      }
      var pathParams = {
        'feature_flag_key': featureFlagKey
      };
      var queryParams = {};
      var headerParams = {};
      var formParams = {};
      var authNames = ['kindeBearerAuth'];
      var contentTypes = [];
      var accepts = ['application/json', 'application/json; charset=utf-8'];
      var returnType = _SuccessResponse["default"];
      return this.apiClient.callApi('/api/v1/environment/feature_flags/{feature_flag_key}', 'DELETE', pathParams, queryParams, headerParams, formParams, postBody, authNames, contentTypes, accepts, returnType, null, callback);
    }

    /**
     * Callback function to receive the result of the deleteEnvironementFeatureFlagOverrides operation.
     * @callback module:api/EnvironmentsApi~deleteEnvironementFeatureFlagOverridesCallback
     * @param {String} error Error message, if any.
     * @param {module:model/SuccessResponse} data The data returned by the service call.
     * @param {String} response The complete HTTP response.
     */

    /**
     * Delete all environment feature flag overrides
     * Delete all environment feature flag overrides.
     * @param {module:api/EnvironmentsApi~deleteEnvironementFeatureFlagOverridesCallback} callback The callback function, accepting three arguments: error, data, response
     * data is of type: {@link module:model/SuccessResponse}
     */
  }, {
    key: "deleteEnvironementFeatureFlagOverrides",
    value: function deleteEnvironementFeatureFlagOverrides(callback) {
      var postBody = null;
      var pathParams = {};
      var queryParams = {};
      var headerParams = {};
      var formParams = {};
      var authNames = ['kindeBearerAuth'];
      var contentTypes = [];
      var accepts = ['application/json', 'application/json; charset=utf-8'];
      var returnType = _SuccessResponse["default"];
      return this.apiClient.callApi('/api/v1/environment/feature_flags/', 'DELETE', pathParams, queryParams, headerParams, formParams, postBody, authNames, contentTypes, accepts, returnType, null, callback);
    }

    /**
     * Callback function to receive the result of the updateEnvironementFeatureFlagOverride operation.
     * @callback module:api/EnvironmentsApi~updateEnvironementFeatureFlagOverrideCallback
     * @param {String} error Error message, if any.
     * @param {module:model/SuccessResponse} data The data returned by the service call.
     * @param {String} response The complete HTTP response.
     */

    /**
     * Update environment feature flag override
     * Update environment feature flag override.
     * @param {String} featureFlagKey The identifier for the feature flag.
     * @param {String} value The override value for the feature flag.
     * @param {module:api/EnvironmentsApi~updateEnvironementFeatureFlagOverrideCallback} callback The callback function, accepting three arguments: error, data, response
     * data is of type: {@link module:model/SuccessResponse}
     */
  }, {
    key: "updateEnvironementFeatureFlagOverride",
    value: function updateEnvironementFeatureFlagOverride(featureFlagKey, value, callback) {
      var postBody = null;
      // verify the required parameter 'featureFlagKey' is set
      if (featureFlagKey === undefined || featureFlagKey === null) {
        throw new Error("Missing the required parameter 'featureFlagKey' when calling updateEnvironementFeatureFlagOverride");
      }
      // verify the required parameter 'value' is set
      if (value === undefined || value === null) {
        throw new Error("Missing the required parameter 'value' when calling updateEnvironementFeatureFlagOverride");
      }
      var pathParams = {
        'feature_flag_key': featureFlagKey
      };
      var queryParams = {
        'value': value
      };
      var headerParams = {};
      var formParams = {};
      var authNames = ['kindeBearerAuth'];
      var contentTypes = [];
      var accepts = ['application/json', 'application/json; charset=utf-8'];
      var returnType = _SuccessResponse["default"];
      return this.apiClient.callApi('/api/v1/environment/feature_flags/{feature_flag_key}', 'PATCH', pathParams, queryParams, headerParams, formParams, postBody, authNames, contentTypes, accepts, returnType, null, callback);
    }
  }]);
  return EnvironmentsApi;
}();
exports["default"] = EnvironmentsApi;
